Automattic/wp-calypso
# Plans: Add more context when a plan and/or feature is selected

Fixes https://github.com/Automattic/wp-calypso/issues/75728

## Proposed Changes

Before redesigning the plans page, we used to provide more context when a single plan and/or feature is "selected" with the `plan` and `feature` query params. You can check this by going to https://wordpress.com/plans?plan=value_bundle&feature=advanced-design-customization&flags=-onboarding%2F2023-pricing-grid.

Looks like we missed to replicate that in the new plans page, so this PR brings back these improvements:
- Show a "Suggested" label on the plan that has been selected.
- Show an additional header when some plans are hidden, so users can know there are more plans available.
- Highlight the selected feature.

**Main grid**
Desktop | Mobile 
--- | ---
<img width="987" alt="Screenshot 2023-05-11 at 13 32 31" src="https://github.com/Automattic/wp-calypso/assets/1233880/07f3542f-b950-4f95-b918-69253eca6d87"> | <img width="589" alt="Screenshot 2023-05-11 at 13 32 18" src="https://github.com/Automattic/wp-calypso/assets/1233880/5b3f9477-6215-4e81-a747-62a352b71822">

**Comparison grid**
Desktop | Mobile
--- | ---
<img width="985" alt="Screenshot 2023-05-11 at 13 36 11" src="https://github.com/Automattic/wp-calypso/assets/1233880/42059da3-a1f9-4468-bce0-ce60845a9ac5"> | Unchanged, it's difficult to highlight a feature here.


## Testing Instructions

- Use the Calypso live link below.
- Go to `/plans/<SITE_DOMAIN>?plan=value_bundle&feature=style-customization`
- Make sure the Premium plan is labeled as "Suggested" instead of "Popular".
- Make sure the "Style customization" feature is highlighted.
- Make sure there is a header indicating that an upgrade is required to access the selected feature.
- Make sure there is a link to browse all plans.
